tasks/copy_to_glacier/test/test_handler.py

```diff
@@ -54,15 +58,420 @@ def test_4_get_bucket(self):
+        collection_url_path = uuid.uuid4().__str__()
+        content_type = uuid.uuid4().__str__()
+        source_key = f'file-staging/blah/{collection_name}__{collection_version}/{filename}'
+        granule_url = f"s3://{source_bucket_name}/{source_key}"
+
+        # todo: use 'side_effect' to verify args. It is safer, as current method does not deep-copy args
+        boto3.client = Mock()
+        s3_cli = boto3.client('s3')
+        s3_cli.copy = Mock(side_effect=[None])
+        s3_cli.head_object = Mock(return_value={'ContentType': content_type})
+
+        event = {
+            'input': [granule_url],
+            'config': {
+                CONFIG_COLLECTION_KEY: {
+                    COLLECTION_NAME_KEY: collection_name,
+                    COLLECTION_VERSION_KEY: collection_version,
+                    'files': [
+                        {
+                            'regex': '[a]+',
+                            'bucket': 'bad_bucket'
+                        },
+                        {
+                            'regex': '^test*.',
+                            'bucket': source_bucket_name
+                        }
+                    ],
+                    COLLECTION_URL_PATH_KEY: collection_url_path
+                },
+                CONFIG_BUCKETS_KEY: {
+                    'bad_bucket': {
+                        'name': 'not_glacier'
+                    },
+                    'glacier': {
+                        'name': destination_bucket_name
+                    }
+                }
+            }
+
+        }
+
+        result = task(event, None)
+
+        self.assertEqual(event['input'], result['input'])
+        granules = result['granules']
+        self.assertIsNotNone(granules)
+        granules = list(granules)
+        self.assertEqual(1, len(granules))
+        granule = granules[0]
+        self.assertEqual(filename, granule['granuleId'])
+        self.assertEqual(1, len(granule['files']))
+        file = granule['files'][0]
+        self.assertEqual(f"{collection_name}__{collection_version}", file['path'])
+        self.assertEqual(f"{collection_name}__{collection_version}", file['url_path'])
+        self.assertEqual(source_bucket_name, file['bucket'])
+        self.assertEqual(granule_url, file['filename'])
+        self.assertEqual(granule_url, file['name'])
+
+        s3_cli.head_object.assert_called_once_with(Bucket=source_bucket_name, Key=source_key)
+        s3_cli.copy.assert_has_calls([call(
+            {'Bucket': source_bucket_name, 'Key': source_key},
+            destination_bucket_name,
+            f"{collection_url_path}/{filename}",
+            ExtraArgs={
+                'StorageClass': 'GLACIER',
+                'MetadataDirective': 'COPY',
+                'ContentType': content_type
+            }
+        )])
+
+    def test_task_bucket_not_present(self):
+        """
+        Bucket not present in collection, should default to 'public' in output.
```

Comment:
Good description. I was just about to ask about this before seeing the above line =).